dame_flame/dame_algorithm.py

```diff
@@ -187,6 +194,19 @@ def algo1(df_all, treatment_column_name = "T", weight_array = [],
     h = 1 # The iteration number
     prev_iter_num_unmatched = len(df_unmatched) # this is for output progress
     
+    if verbose==3:
+        print("Iteration number: ", h)
+        print("Matched groups formed: ", len(units_in_g))
+        unmatched_treated = df_unmatched[treatment_column_name].sum()
+        unmatched_control = len(df_unmatched) - unmatched_treated             
+        total_treated = df_all[treatment_column_name].sum()
+        print("Unmatched treated units: ", unmatched_treated,"out of a total of ", total_treated, "treated units .")
+        print("Unmatched control units: ", unmatched_control,"out of a total of ", orig_len_df_all-total_treated, 
+                  "control units")
+        print("Number of matches made so far: ", orig_len_df_all - len(df_unmatched))
+
+    
+    
```

Comment:
I do think you found the right place to put this code. But, since I'm now realizing this is repeated code here, in the loop for the output, and also in the flame function....could you make a helper function (perhaps in flame_dame_helpers.py) called something like "print_verbose_output" and send it all of the relevant paramaters? I am just trying to minimize repetitive code and keep the files as brief as possible.